src/menu/menu.service.ts

```diff
@@ -1,26 +1,121 @@
-import { Injectable } from '@nestjs/common';
+import { MenuRepository } from './menu.repository';
+import {
+  Injectable,
+  BadRequestException,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
 import { CreateMenuDto } from './dto/create-menu.dto';
 import { UpdateMenuDto } from './dto/update-menu.dto';
+import { CategoriesService } from 'src/categories/categories.service';
+import { Menu } from './entities/menu.entity';
 
 @Injectable()
 export class MenuService {
-  create(createMenuDto: CreateMenuDto) {
-    return 'This action adds a new menu';
+  constructor(
+    @InjectRepository(MenuRepository) private menuRepository: MenuRepository,
+    private categoriesService: CategoriesService,
+  ) {}
+
+  async create(createMenuDto: CreateMenuDto) {
+    const { icon, text, category_id } = createMenuDto;
+
+    const category = await this.categoriesService.findOne(category_id);
+
+    const href = category.text.split(' ').join('-').toLocaleLowerCase();
+
+    const menuSaved = await this.menuRepository.save({
+      href,
+      icon,
+      text,
+      category,
+    });
+
+    if (!menuSaved) {
+      throw new BadRequestException();
```

Comment:
It should return an explicit error message or error code for better understanding about the error